pkg/feature/raw_resources.go

```diff
@@ -105,3 +102,50 @@ func patchUsingMergeStrategy(ctx context.Context, cli client.Client, patch *unst
 
 	return nil
 }
+
+// At this point we only look at what is defined for the resource in the desired state (source),
+// so the one provided by the operator (e.g. embedded manifest files).
+//
+// Although the actual instance on the cluster (the target) might be in a different state,
+// we intentionally do not address this scenario due to the lack of clear requirements
+// on the extent to which users can modify it. Additionally, addressing this would
+// necessitate extra reporting (e.g., via status.conditions on a FeatureTracker)
+// to highlight discrepancies between the actual and desired state when users opt out
+// of operator management/reconciliation.
+func shouldReconcile(source *unstructured.Unstructured) bool {
+	if isManaged(source) {
+		return true
+	}
+
+	if isUnmanaged(source) {
+		return false
+	}
+
+	// In all the other cases preserve original behaviour
+	return false
+}
+
+func markAsManaged(objs []*unstructured.Unstructured) {
+	for _, obj := range objs {
+		objAnnotations := obj.GetAnnotations()
+		if objAnnotations == nil {
+			objAnnotations = make(map[string]string)
+		}
+
+		// If resource already has management mode defined, it should take precedence
+		if _, exists := objAnnotations[annotations.ManagedByODHOperator]; !exists {
+			objAnnotations[annotations.ManagedByODHOperator] = "true"
+			obj.SetAnnotations(objAnnotations)
+		}
+	}
+}
+
+func isUnmanaged(obj *unstructured.Unstructured) bool {
+	managed, isDefined := obj.GetAnnotations()[annotations.ManagedByODHOperator]
+	return isDefined && managed == "false"
+}
+
+func isManaged(obj *unstructured.Unstructured) bool {
+	managed, isDefined := obj.GetAnnotations()[annotations.ManagedByODHOperator]
+	return isDefined && managed == "true"
+}
```

Comment:
So, how is the 3rd state, !isDefined, which is neither managed nor unmanaged called? :)

Reply:
then `shouldReconcile`() return false in the case of !IsDefined , right?

Reply:
> then `shouldReconcile`() return false in the case of !IsDefined , right?

Yes. As far as I understand https://github.com/opendatahub-io/opendatahub-operator/pull/1103#discussion_r1670949924 you discussed it offline and kept it in purpose. It just looks a bit weird from the reader's prospective :)

In general, as mentioned in the comment, shouldReconcile is just isManaged(), and Unmanaged probably means !isManaged().

Reply:
I admit my 🧠 was tripping at the time of writing it. Happy to discuss how to make it better.